### DISCUSSION TITLE
Public player DOM modifications by Javascript

### DISCUSSION THREAD

**Discussion by krono-i2** [AzuraCast/AzuraCast#5845] [Q&A]
Hello!
I'm trying to modify the beheviour of the public player to match my requirements. In order to obtain my needing I'm using custom javascript and CSS.
Specifically, into the `branding` page I'm insert the following javascript.
```
if (document.URL === "https://radio.rcsnetwork.org/public/rcsnetwork") {
  const newStyle = document.createElement('link');
  newStyle.setAttribute('rel','stylesheet');
  newStyle.setAttribute('type','text/css');
  newStyle.setAttribute('href','https://example.org/styles/azuracast-custom-player.css');
  document.head.appendChild(newStyle);
  
  const newScript = document.createElement('script');
  newScript.setAttribute('type','text/javascript');
  newScript.setAttribute('src','https://example.org/app/scripts/azuracast-custom-player.js');
  const nonce = document.getElementsByTagName('script')[25].nonce;
  newScript.setAttribute('nonce',nonce);
  document.body.appendChild(newScript);
}
```
The tags `link` and `script` are correctly loaded into the webpage.
The problem is that when my `azuracast-custom-player.js` trys to modify the DOM, sometimes it does not find the elements to be modified. So, I found out the DOM is dinamically created by the player script.
I can solve the problem by introducing a simple delay, but I'm searching for a solution to run my script after the player's elements are completly created.
Thank you for helping.
Ragards.

Ivan

**Comment by Moonbase59** [comment]
Can’t you use something on the lines of `load` and/or `DOMContentLoaded`?

Like in
```js
document.addEventListener("DOMContentLoaded", function(event){
  // your code here
});
```

The `DOMContentLoaded` should be supported in almost every browser nowadays. As far as I remember, it’s fired when the DOM is complete but _before_ all dependencies are loaded.

If you need that, you might go for the `load`event:
```js
window.addEventListener('load', function () {
  // your code here
})
```

Good luck, and let us know.
